src/QMCDrivers/WFOpt/QMCCostFunctionBatched.cpp

```diff
@@ -272,38 +270,40 @@ void QMCCostFunctionBatched::checkConfigurations()
 
   // Create crowd-local storage for evaluation
   outputManager.pause();
-  opt_eval_.resize(opt_num_crowds_);
-  for (int i = 0; i < opt_num_crowds_; i++)
-    opt_eval_[i] = std::make_unique<CostFunctionCrowdData>(opt_batch_size_, W, Psi, H, H_KE_node_names_, *MoverRng[0]);
+  size_t opt_num_crowds = walkers_per_crowd_.size();
+  opt_eval_.resize(opt_num_crowds);
+  for (int i = 0; i < opt_num_crowds; i++)
+    opt_eval_[i] =
+        std::make_unique<CostFunctionCrowdData>(walkers_per_crowd_[i], W, Psi, H, H_KE_node_names_, *MoverRng[0]);
   outputManager.resume();
 
 
   // Divide samples among the crowds
-  std::vector<int> samples_per_crowd(opt_num_crowds_ + 1);
-  FairDivide(rank_local_num_samples_, opt_num_crowds_, samples_per_crowd);
+  std::vector<int> samples_per_crowd(opt_num_crowds + 1);
+  FairDivide(rank_local_num_samples_, opt_num_crowds, samples_per_crowd);
 
   // lambda to execute on each crowd
   auto evalOptConfig = [](int crowd_id, UPtrVector<CostFunctionCrowdData>& opt_crowds,
-                          std::vector<int>& samples_per_crowd, int crowd_size, std::vector<ParticleGradient*>& gradPsi,
-                          std::vector<ParticleLaplacian*>& lapPsi, Matrix<Return_rt>& RecordsOnNode,
-                          Matrix<Return_rt>& DerivRecords, Matrix<Return_rt>& HDerivRecords, const SampleStack& samples,
-                          opt_variables_type& optVars, bool needGrads, bool compute_nlpp,
-                          const std::string& includeNonlocalH) {
+                          std::vector<int>& samples_per_crowd, std::vector<Index>& walkers_per_crowd,
+                          std::vector<ParticleGradient*>& gradPsi, std::vector<ParticleLaplacian*>& lapPsi,
+                          Matrix<Return_rt>& RecordsOnNode, Matrix<Return_rt>& DerivRecords,
+                          Matrix<Return_rt>& HDerivRecords, const SampleStack& samples, opt_variables_type& optVars,
+                          bool needGrads, bool compute_nlpp, const std::string& includeNonlocalH) {
     CostFunctionCrowdData& opt_data = *opt_crowds[crowd_id];
 
     int local_samples = samples_per_crowd[crowd_id + 1] - samples_per_crowd[crowd_id];
     int num_batches;
     int final_batch_size;
 
-    compute_batch_parameters(local_samples, crowd_size, num_batches, final_batch_size);
+    compute_batch_parameters(local_samples, walkers_per_crowd[crowd_id], num_batches, final_batch_size);
 
     for (int inb = 0; inb < num_batches; inb++)
     {
-      int curr_crowd_size = crowd_size;
+      int curr_crowd_size = walkers_per_crowd[crowd_id];
```

Comment:
If this is a convenient variable, let us add a const.

Reply:
This variable can't be const because it may get changed in the following lines.

Reply:
I see. Then leave it not const. Could you rename this one as `this_batch_size`?

Reply:
How about `current_batch_size`?  I don't like using `this` in variable names because of the special meaning of "this" in C++.

Reply:
This is an example of surprise. Let me explain.
`curr_crowd_size` is somehow a const once its value is settled. So it is possible to made a decision and assign its value once and mark it const. It is better than set an initial value and then modify it later. Although the current way looks OK, the modification is on the second line but we do lost the const protection.

Reply:
`current_batch_size` is good

Reply:
The ternary operator would be one solution.
```
const int current_batch_size = inb == num_batches - 1 ? final_batch_size : walkers_per_crowd[crowd_id
```

Reply:
That is a clean solution that we can go for.